### ISSUE TITLE
Add support for +named-route-compoents.page.svelte to make finding a specific route file easier

### ISSUE THREAD

**Issue by aleksilassila** [sveltejs/kit#10331]
### Describe the problem

When a project grows larger, there are a lot of files with similar names such as +page.svelte or +page.ts in src/routes/:
```
src/routes/
   contact/
      +page.svelte
   about/
      +page.svelte
   landing/
      +page.svelte
   newsletter/
      +page.svelte
   +page.svelte
```

This makes it a pain to:
a) navigate to these files by name. If every file has the same name, you cannot search for them. The only way to find these files is to click through the file tree, which is cumbersome, especially in large projects.
b) Managing your tabs in an IDE. If there's a +page.svelte open on your editor as one of multiple tabs and you have been working on multiple route files recently, you have to read the contents of the file to figure out which route's +page.svelte file it is.

### Describe the proposed solution

Allow prepending a custom name to each file that will be ignored by the compiler. For exmaple, instead of:
```
src/routes/
   contact/
      +page.svelte
      +page.ts
   about/
      +page.svelte
      +page.ts
   landing/
      +page.svelte
      +page.ts
   newsletter/
      +page.svelte
   +page.svelte
```
you would have:
```
src/routes/
   contact/
      +contact.page.svelte
      +page.ts <--- it would be optional and named +page files would be treated as if they were just +page.sveltes
   about/
      +about.page.svelte
      +whatever.page.ts
   landing/
      +landing.page.svelte
      +landing.page.ts
   newsletter/
      +newsletter.page.svelte
   +home.page.svelte
```

### Alternatives considered

Make IDE plugins such as Svelte extension for vs code automatically search and display files with common names such as +page.svelte as if the parent directory was a part of the file name. Like what's already being done with file tab names when you have two files with same names open at the same time, but do this regardless of how many +page.svelte or +page.ts files you have open.

### Importance

would make my life easier

**Comment by dummdidumm** [comment]
One way which would be consistent with grouped routes would be to make it `+page(name-here).svelte`. Folders in `(braces)` are ignored for the url path, so we could leverage similar semantics for this.

**Comment by ghostdevv** [comment]
IMO supporting something like this would add more noise to the file system, which is already grouped by folders which allow for semantic naming with `(folders)` in brackets. I think enabling things like `"workbench.editor.labelFormat": "medium"` in vscode is the best solution to confusion when you have multiple `+page...` files open

**Comment by mustofa-id** [comment]
> you cannot search for them

![image](https://github.com/sveltejs/kit/assets/25121822/e7912e73-a185-4d94-9a8b-4de7f35f0932)

**Comment by aleksilassila** [comment]
> > you cannot search for them
> 
> ![image](https://user-images.githubusercontent.com/25121822/251686505-e7912e73-a185-4d94-9a8b-4de7f35f0932.png)

Fair point, you can search for them but it appears that the directory hits are always below file name matches, which there are in some cases a lot of. Also always having to type +page results in more typing.

Though if you do the things pointed in above comments, this becomes more of an minor inconvenience. Maybe solving the issue with better IDE support would be a better way of going about this, although I'm not sure if what I described in the alternative solution can be done in an IDE extension or a plugin.

**Comment by MrGerbeck** [comment]
As someone who doesn't mind the +page format, I think the proposal makes sense. It would be optional and the perceived noise by some doesn't matter much as it is an opt-in.

**Comment by protibimbok** [comment]
How about exporting a variable?
```typescript
export const ROUTE_NAME = 'name_goes_here'
```
As the sveltekit already builds dynamic `$types` it should not be a problem.

**Comment by dshook** [comment]
> One way which would be consistent with grouped routes would be to make it `+page(name-here).svelte`. Folders in `(braces)` are ignored for the url path, so we could leverage similar semantics for this.

Why not just the name between dots?  `+page.WhateverYouWant.svelte`

Setting VS code `"workbench.editor.labelFormat": "medium"` doesn't work for me when you get to a route that's 5 levels nested.  The titles are way too massive and cluttering.

**Comment by WBSpmichal** [comment]
I had an idea, feel free to ignore if it's not feasible with the current architecture. 

My question is, do the files HAVE to end with .svelte? what if, for example, the file ending was used to identify the file function? Example, +file-name.page would be the page component, +file-name.server would be for API routes, etc. 

+file-name.page.svelte would also be fine, although getting a bit wordy.

**Comment by webJose** [comment]
@WBSpmichal they have to be .svelte or the svelte pre-processor would need to be reconfigured.

I have seen conversations about this.  I know what I'll say is specific to VS Code, but this is what most people use anyway:  Ctrl + P, then type "+".  Now you see the files with their paths.  Super simple.

**Comment by mustofa-id** [comment]
I found out that VSCode can do something like this:

```json
"workbench.editor.customLabels.patterns": {
  "**/routes/**/+{page,layout,error,server}.{svelte,js,ts,server.ts,server.js}": "${dirname}/${filename}.${extname}"
}
```

![image](https://github.com/user-attachments/assets/d6fff834-434d-4cb1-9167-40132fde1f76)

source: [Reddit](https://www.reddit.com/r/nextjs/comments/1ey6h7r/comment/ljdz64n)

**Comment by dominikg** [comment]
there are some downsides to allowing optional "tags" in filenames. 

- adds complexity in parsing the route structure
- increases verbosity in filenames
- limits our ability to add features that need extra information in the filename

With the tools offered by ides to display the parent directory name and find dialogs displaying paths too i don't think the cost/benefit ratio is favorable here.
